packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/emr/integ.emr-create-cluster-add-step-runtime-role.ts

```diff
@@ -0,0 +1,126 @@
+
+/*
+ * Create a state machine with an EMR cluster and adds a step that uses a runtime role.
+ *
+ * PREREQUISITES:
+ * 1. Fill in the instances block for EmrCreateCluster
+ * 2. Create the EMR security configuration: https://aws.amazon.com/blogs/big-data/introducing-runtime-roles-for-amazon-emr-steps-use-iam-roles-and-aws-lake-formation-for-access-control-with-amazon-emr/
+ * 3. Add the security configuration to EmrCreateCluster
+ * 4. Update the runtime role's trust policy as outlined below
+ *
+ * Stack verification steps:
+ * The generated State Machine can be executed from the CLI (or Step Functions console)
+ * and runs with an execution status of `Succeeded`.
+ *
+ * -- aws stepfunctions start-execution --state-machine-arn <state-machine-arn-from-output> provides execution arn
+ * -- aws stepfunctions describe-execution --execution-arn <state-machine-arn-from-output> returns a status of `Succeeded`
+ */
+
+const app = new App();
+const stack = new Stack(app, 'aws-cdk-emr-add-step-runtime-role');
+
+const createClusterStep = new tasks.EmrCreateCluster(stack, 'EmrCreateCluster', {
+  instances: {
+    /*
+    instanceFleets: [
+      {
+        instanceFleetType: tasks.EmrCreateCluster.InstanceRoleType.MASTER,
+        instanceTypeConfigs: [
+          {
+            instanceType: 'm5.xlarge',
+          },
+        ],
+        targetOnDemandCapacity: 1,
+      },
+    ],
+    ec2SubnetId: 'subnet-xxxxxx',
+    */
+  },
+  name: 'Cluster',
+  releaseLabel: 'emr-6.13.0',
+  integrationPattern: sfn.IntegrationPattern.RUN_JOB,
+  tags: {
+    Key: 'Value',
+  },
+  //securityConfiguration: 'security-configuration',
+  applications: [
+    {
+      name: 'Spark',
+    },
+  ],
+});
+
+const executionRole = new iam.Role(stack, 'Role', {
+  roleName: 'EmrStepExecutionRole',
+  assumedBy: new iam.ArnPrincipal(createClusterStep.clusterRole.roleArn),
+});
+
+/*
+The trust policy of the runtime execution role needs to have the following:
+@see https://aws.amazon.com/blogs/big-data/introducing-runtime-roles-for-amazon-emr-steps-use-iam-roles-and-aws-lake-formation-for-access-control-with-amazon-emr/
+{
+  "Version": "2012-10-17",
+  "Statement": [
+      {
+          "Effect": "Allow",
+          "Principal": {
+              "AWS": "arn:aws:iam::<AWS_ACCOUNT_ID>:role/<CLUSTER_ROLE>"
+          },
+          "Action": "sts:AssumeRole"
+      },
+      {
+          "Effect": "Allow",
+          "Principal": {
+              "AWS": "arn:aws:iam::<AWS_ACCOUNT_ID>:role/<CLUSTER_ROLE>"
+          },
+          "Action": "sts:SetSourceIdentity"
+      },
+      {
+          "Effect": "Allow",
+          "Principal": {
+              "AWS": "arn:aws:iam::<AWS_ACCOUNT_ID>:role/<CLUSTER_ROLE>"
+          },
+          "Action": "sts:TagSession",
+          "Condition": {
+              "StringEquals": {
+                  "aws:RequestTag/LakeFormationAuthorizedCaller": "Amazon EMR"
+              }
+          }
+      }
+  ]
+}
+*/
```

Comment:
```suggestion

const executionRolePrincipal = new iam.ArnPrincipal(createClusterStep.clusterRole.roleArn);
const executionRole = new iam.Role(stack, 'Role', {
  roleName: 'EmrStepExecutionRole',
  assumedBy: executionRolePrincipal,
});
executionRole.assumeRolePolicy?.addStatements(
    new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      principals: [
        executionRolePrincipal,
      ],
      actions: [
        'sts:SetSourceIdentity',
      ],
    }),
    new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      principals: [
        executionRolePrincipal,
      ],
      actions: [
        'sts:TagSession',
      ],
      conditions: {
        StringEquals: {
          'aws:RequestTag/LakeFormationAuthorizedCaller': 'Amazon EMR',
        },
      },
    }),
  );
```
This should generate the necessary policy.